pulsar-client-cpp/include/pulsar/ConsumerConfiguration.h

```diff
@@ -33,6 +33,7 @@ class PulsarWrapper;
 
 /// Callback definition for non-data operation
 typedef boost::function<void(Result result)> ResultCallback;
+typedef boost::function<void(const Message& msg)> ReceiveCallback;
```

Comment:
it's a async-receive and it will never have any failure as a result and it only returns a message when available so, I think there is no need of result in this case.